server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -421,6 +422,8 @@ static void validateTypeName(String type) {
             MapperMergeValidator.validateFieldReferences(fieldMappers, fieldAliasMappers,
                 fullPathObjectMappers, fieldTypes);
 
+            GeoContextMapping.validateGeoContextPaths(indexSettings.getIndexVersionCreated(), fieldMappers, fieldTypes::get);
```

Comment:
Because this method provides a general way to valid context references, I wonder if it makes sense to future-proof a bit and reference `ContextMapping` instead, and also rename `validateGeoContextPaths` to `validateContextPaths`.